### PR TITLE
API doc restructuring

### PR DESCRIPTION
**Pull request by hpohekar**
image

`--------------------------------------------------------------------------------------------------------------------`

1. Updated `launcher.rst`.

image

`--------------------------------------------------------------------------------------------------------------------`

3.Updated `utils.rst`.
4. Updated `tuigen.py`.
5. Updated `datamodelgen.py`.
6. Updated `settings_rstgen.py`

### CODE REVIEW

**Review by prmukherj — APPROVED** [review]

File: doc/Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ help:
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 %: Makefile
-	@python settings_rstgen.py
+#	@python settings_rstgen.py
```
Comment (dnwillia-work) [inline doc/Makefile:19]: remove?

File: doc/make.bat [hunk 2]
```diff
@@ -29,7 +29,7 @@ if errorlevel 9009 (
 	exit /b 1
 )
 
-python settings_rstgen.py
+REM python settings_rstgen.py
```
Comment (dnwillia-work) [inline doc/make.bat:32]: Remove?

File: doc/source/api/index.rst [hunk 3]
```diff
@@ -9,29 +9,29 @@ Here, you can find the application programming interfaces for
 such things as launching Fluent, assorted utilities, as well as 
 the interfaces for the meshing and solver components of Fluent. 
 
-Launching Fluent
-----------------
+:ref:`ref_launcher_launcher`
+-----------------------------
```
Comment (dnwillia-work) [inline doc/source/api/index.rst:13]:
```suggestion
----------------------------
```

File: doc/source/api/index.rst [hunk 4]
```diff
@@ -9,29 +9,29 @@ Here, you can find the application programming interfaces for
...
 
 This component provides access to starting Fluent locally in 
 server mode or connecting to a running Fluent server instance.
 
-PyFluent utility functions
---------------------------
+:ref:`ref_utils`
+-----------------
```
Comment (dnwillia-work) [inline doc/source/api/index.rst:19]:
```suggestion
----------------
```

File: doc/source/api/index.rst [hunk 5]
```diff
@@ -9,29 +9,29 @@ Here, you can find the application programming interfaces for
...
 
 This component consists solely of a function to allow for 
 asynchronous execution.
 
-Meshing
--------
+:ref:`ref_meshing`
+-------------------
```
Comment (dnwillia-work) [inline doc/source/api/index.rst:25]:
```suggestion
------------------
```

File: doc/make.bat [hunk 6]
```diff
@@ -11,7 +11,7 @@ if "%SPHINXBUILD%" == "" (
 )
 set SOURCEDIR=source
 set BUILDDIR=_build
-set SPHINXOPTS=-j auto -W --keep-going -w build_errors.txt -N -q
+set SPHINXOPTS=-j auto -W --keep-going -w build_errors.txt -N -v
```
Comment (dnwillia-work) [inline doc/make.bat:14]: I think we were using `-q` here before to suppress some of the build output coming from the Fluent RPC calls which clutter things up when you are trying to find problems with the documentation (eg: warnings etc..)

File: doc/Makefile (hunk 1 above)
Comment (hpohekar) [reply doc/Makefile:19]: Currently due to `:recursive:` flag the `rst` files generated for `settings` are near around 70,000 so it takes more than 20 hrs locally just to read `rst` files. So we commented it out now.

First we want to ensure that the `datamodel` and `tui` works fine then we will improve the performance for `settings`.

File: doc/make.bat (hunk 2 above)
Comment (hpohekar) [reply doc/make.bat:32]: The GitHub is failing while reading `settings` rst files hence we have commented this for now.

We are going to uncomment this later on.

File: doc/make.bat (hunk 6 above)
Comment (hpohekar) [reply doc/make.bat:14]: Yes, that's right. but as mentioned above to see the progress and all warnings we have changed it to `-v`. The warnings are helpful to resolve the documentation issues.

We will change this to `-q` again.

File: doc/source/api/index.rst (hunk 3 above)
Comment (hpohekar) [reply doc/source/api/index.rst]: Done. Thank you.

File: doc/source/api/index.rst (hunk 4 above)
Comment (hpohekar) [reply doc/source/api/index.rst]: Done. Thank you.

File: doc/source/api/index.rst (hunk 5 above)
Comment (hpohekar) [reply doc/source/api/index.rst]: Done. Thank you.

File: doc/source/api/general/rpvars.rst [hunk 7]
```diff
@@ -0,0 +1,13 @@
+.. _ref_rpvars:
+
+rpvars
```
[the remark this answers is not in the document]
Comment (hpohekar) [reply doc/source/api/general/rpvars.rst:3]: @mkundu1 Thank you very much.

[the remark this answers is not in the document]
Comment (hpohekar) [reply doc/source/api/general/rpvars.rst:3]: Done. Thank you.

**Comment by mkundu1** [comment]
@hpohekar I'm not sure why the pre-commit issues are coming. Can you update with the latest main branch? There should not be any issue in the pre-commit as it works fine in the main branch (pre-commit checks just the source code and docstrings - what is committed in git).

**Comment by hpohekar** [comment]
> @hpohekar I'm not sure why the pre-commit issues are coming. Can you please update with the latest main branch? There should not be any issue in the pre-commit as it works fine in the main branch (pre-commit checks just the source code and docstrings - what is committed in git).

Yes, sure, I will try again. I have tried several ways to handle this but it's strange that it is failing in this branch. Thank you.

**Comment by hpohekar** [comment]
> @hpohekar I'm not sure why the pre-commit issues are coming. Can you please update with the latest main branch? There should not be any issue in the pre-commit as it works fine in the main branch (pre-commit checks just the source code and docstrings - what is committed in git).

@mkundu1 My local branch is already up to date with main branch.

**Comment by mkundu1** [comment]
@hpohekar Why the underlines below `Parameters`/`Returns` are removed from datamodel_*.py docstrings? The underlines follow [numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html).

**Comment by hpohekar** [comment]
> @hpohekar Why the underlines below `Parameters`/`Returns` are removed from datamodel_*.py docstrings? The underlines follow [numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html).

To avoid `unexcepted title or transition error`

**Comment by mkundu1** [comment]
> > @hpohekar Why the underlines below `Parameters`/`Returns` are removed from datamodel_*.py docstrings? The underlines follow [numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html).
> 
> To avoid `unexcepted title or transition error`

I've a fix for the title error in doc/issues branch.

File: doc/make.bat (hunk 2 above)
Comment (dnwillia-work) [reply doc/make.bat]: Yeah, same as above.  I'm not sure the rst generation should be done here right.  I think for the TUI it's happening elsewhere, as part of the calls in `codegen`.

Comment (hpohekar) [reply doc/make.bat]: @dnwillia-work 

We have verified that we are able to generate `datamodel`, `TUI` and `settings` html docs locally without any error.
We have updated the code and reverted above change. Thank you.

image

`---------------------------------------------------------------------------------------------------------------------`

image

`---------------------------------------------------------------------------------------------------------------------`

image

File: doc/Makefile (hunk 1 above)
Comment (hpohekar) [reply doc/Makefile]: We have verified that we are able to generate datamodel, TUI and settings html docs locally without any error.
We have updated the code and reverted above change. Thank you.

File: doc/source/conf.py [hunk 8]
```diff
@@ -176,7 +202,6 @@ def _stop_fluent_container(gallery_conf, fname):
     },
     "navbar_end": ["version-switcher", "theme-switcher", "navbar-icon-links"],
     "navigation_depth": -1,
-    "collapse_navigation": True,
```
Comment (mkundu1) [inline doc/source/conf.py:179]: @hpohekar I think this one was removed by some merges. This option was recently added to have some speedup (#1331) in the doc build.

Comment (hpohekar) [reply doc/source/conf.py]: We have added it now. Thank you.

File: doc/Makefile [hunk 9]
```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -j auto -W --keep-going -w build_errors.txt -N -q
+SPHINXOPTS    = -j auto --keep-going -w build_errors.txt -N -q
```
Comment (dnwillia-work) [inline doc/Makefile:5]: Did you mean to turn off -W?

Comment (hpohekar) [reply doc/Makefile:5]: Yes, we are getting more than 25000 warnings thus it hangs so to avoid this we have removed `-W` option.

**Comment by hpohekar** [comment]
Due unresolved `pre-commit.ci - pr` error we have implemented same changes in https://github.com/pyansys/pyfluent/pull/1363

**Pull request closed without merging**